app/controllers/api/users_controller.rb

```diff
@@ -9,10 +9,11 @@ def index
     @tag = params[:tag]
     @target = params[:target]
     @target = 'student_and_trainee' unless target_allowlist.include?(@target)
+    @watch = params[:watch]
 
     target_users =
       if @target == 'followings'
-        followings = Following.where(follower_id: current_user.id).select('followed_id')
+        followings = current_user.following_list(@watch)
         User.where(id: followings)
```

Comment:
【MUST】
`following_list` という名前でidを返すのは違和感があるのでなんとかしたいですね
userのまとまりを直接返した方がスッキリすると思います

```suggestion
        current_user.following_list(watch: @watch)
```

`following_list` メソッドの中身はこんな感じでどうでしょう

```ruby
def following_list(watch: '')
  if %w[true false].include?(watch)
    following.includes(:passive_relationships).where(followings: { watch: watch })
  else
    following
  end
end
```

Reply:
> following_list という名前でidを返すのは違和感があるのでなんとかしたいですね
userのまとまりを直接返した方がスッキリすると思います

確かにそうですね！
提案していただいた`following_list`で修正しました。
`following.includes(:passive_relationships).where(followings: { watch: watch })`は自分には思い付かなかったので参考になりました！

`app/controllers/api/users_controller.rb`だけでなく
`app/controllers/users_controller.rb`の方も修正しました。